### DISCUSSION TITLE
cant understand this holding registers

### DISCUSSION THREAD

**Discussion by majungman** [pymodbus-dev/pymodbus#959] [Q&A] [answered]
### Versions

* Python:
* OS:
* Pymodbus:
* Modbus Hardware (if used): 

### Pymodbus Specific
* Server: tcp/rtu/ascii - sync/async
* Client: tcp/rtu/ascii - sync/async

### Description
hi,  i am trying to read the power meter value through modbus tcp/ip 
so i connect to the controller ip port 502 

Slave ID, 1, (0x01)
Function, Read Holding Registers, (0x03)
Starting Address, 40001, (0x00 0x00)
Quantity, 120, (0x00 0x78)

RX (Response)
01 03 F0 12 34 0F FF 00 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 50 6F 77 65 72 20 4D 65 74 65 72 20 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 50 4D 35 33 31 30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 53 63 68 6E 65 69 64 65 72 20 45 6C 65 63 74 72 69 63 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 3B A8 00 1B 00 00 00 00 00 00 16 9D 00 00 00 00 00 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00 80 00
Slave ID, 1, (0x01)
Function, Read Holding Registers, (0x03)
40001, 4660, (0x12 0x34)
40002, 4095, (0x0F 0xFF)
40003, 0, (0x00 0x00)
40004, 32768, (0x80 0x00)
40005, 32768, (0x80 0x00)
40006, 32768, (0x80 0x00)
40007, 32768, (0x80 0x00)
40008, 32768, (0x80 0x00)
40009, 32768, (0x80 0x00)
40010, 32768, (0x80 0x00)
40011, 32768, (0x80 0x00)
40012, 32768, (0x80 0x00)
40013, 32768, (0x80 0x00)
40014, 32768, (0x80 0x00)
40015, 32768, (0x80 0x00)
40016, 32768, (0x80 0x00)
40017, 32768, (0x80 0x00)
40018, 32768, (0x80 0x00)
40019, 32768, (0x80 0x00)
40020, 32768, (0x80 0x00)
40021, 32768, (0x80 0x00)
40022, 32768, (0x80 0x00)
40023, 32768, (0x80 0x00)
40024, 32768, (0x80 0x00)
40025, 32768, (0x80 0x00)
40026, 32768, (0x80 0x00)
40027, 32768, (0x80 0x00)
40028, 32768, (0x80 0x00)
40029, 32768, (0x80 0x00)
40030, 20591, (0x50 0x6F)
40031, 30565, (0x77 0x65)
40032, 29216, (0x72 0x20)
40033, 19813, (0x4D 0x65)
40034, 29797, (0x74 0x65)
40035, 29216, (0x72 0x20)
40036, 0, (0x00 0x00)
40037, 0, (0x00 0x00)
40038, 0, (0x00 0x00)
40039, 0, (0x00 0x00)
40040, 0, (0x00 0x00)
40041, 0, (0x00 0x00)
40042, 0, (0x00 0x00)
40043, 0, (0x00 0x00)
40044, 0, (0x00 0x00)
40045, 0, (0x00 0x00)
40046, 0, (0x00 0x00)
40047, 0, (0x00 0x00)
40048, 0, (0x00 0x00)
40049, 0, (0x00 0x00)
40050, 20557, (0x50 0x4D)
40051, 13619, (0x35 0x33)
40052, 12592, (0x31 0x30)
40053, 0, (0x00 0x00)
40054, 0, (0x00 0x00)
40055, 0, (0x00 0x00)
40056, 0, (0x00 0x00)
40057, 0, (0x00 0x00)
40058, 0, (0x00 0x00)
40059, 0, (0x00 0x00)
40060, 0, (0x00 0x00)
40061, 0, (0x00 0x00)
40062, 0, (0x00 0x00)
40063, 0, (0x00 0x00)
40064, 0, (0x00 0x00)
40065, 0, (0x00 0x00)
40066, 0, (0x00 0x00)
40067, 0, (0x00 0x00)
40068, 0, (0x00 0x00)
40069, 0, (0x00 0x00)
40070, 21347, (0x53 0x63)
40071, 26734, (0x68 0x6E)
40072, 25961, (0x65 0x69)
40073, 25701, (0x64 0x65)
40074, 29216, (0x72 0x20)
40075, 17772, (0x45 0x6C)
40076, 25955, (0x65 0x63)
40077, 29810, (0x74 0x72)
40078, 26979, (0x69 0x63)
40079, 0, (0x00 0x00)
40080, 0, (0x00 0x00)
40081, 0, (0x00 0x00)
40082, 0, (0x00 0x00)
40083, 0, (0x00 0x00)
40084, 0, (0x00 0x00)
40085, 0, (0x00 0x00)
40086, 0, (0x00 0x00)
40087, 0, (0x00 0x00)
40088, 0, (0x00 0x00)
40089, 0, (0x00 0x00)
40090, 15272, (0x3B 0xA8)
40091, 27, (0x00 0x1B)
40092, 0, (0x00 0x00)
40093, 0, (0x00 0x00)
40094, 0, (0x00 0x00)
40095, 5789, (0x16 0x9D)
40096, 0, (0x00 0x00)
40097, 0, (0x00 0x00)
40098, 0, (0x00 0x00)
40099, 32768, (0x80 0x00)
40100, 32768, (0x80 0x00)
40101, 32768, (0x80 0x00)
40102, 32768, (0x80 0x00)
40103, 32768, (0x80 0x00)
40104, 32768, (0x80 0x00)
40105, 32768, (0x80 0x00)
40106, 32768, (0x80 0x00)
40107, 32768, (0x80 0x00)
40108, 32768, (0x80 0x00)
40109, 32768, (0x80 0x00)
40110, 32768, (0x80 0x00)
40111, 32768, (0x80 0x00)
40112, 32768, (0x80 0x00)
40113, 32768, (0x80 0x00)
40114, 32768, (0x80 0x00)
40115, 32768, (0x80 0x00)
40116, 32768, (0x80 0x00)
40117, 32768, (0x80 0x00)
40118, 32768, (0x80 0x00)
40119, 32768, (0x80 0x00)
40120, 32768, (0x80 0x00)

this is the reading that i got , 

### Code and Logs

```python
 # code and logs here.

```

**Comment by swamper123** [comment]
On which address starts the power meter value and where does it ends?
Does it realy start at 40001 and ends at 40120?

What does the doc say?

**Reply by majungman** [reply]
thats the issue , i am not sure which address i suppose to be reading , 
so for that i have to refer to the power meter documents ? the meter that im trying to read is schneider electric power meter

**Reply by swamper123** [reply]
I would recommened: Read the docs of your device.
"Modbus register map" is a thing you may want to look for.
And in case of emergency, just call the hotline/write a mail to schneider electric and ask them where to look for.

We can't help you here, since the only thing we can see is: "Yep! This seems to be an successfull read of the holding registers!"

**Comment by janiversen** [comment, marked as answer]
Reading 120 registers in one go, is going to the limit (255 bytes is max.), and will cause problems in a number of devices. Make 2 reads each of 60.

But reading your description it seems there are no problem, it would help if you wrote where you see a problem.

**Comment by janiversen** [comment]
Furthermore I looked over the Schneider implementations I have (including some power meters) and none have 120 holding registers....are you sure your meter have that many holding registers ?

The load profile in the power meter, have much more values, but normally read that with the file command.

**Reply by majungman** [reply]
yes, i guess i was reading it wrong , still trying to making sense of the values . i have got the modbus register mapping doc .
